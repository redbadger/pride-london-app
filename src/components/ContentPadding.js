// @flow
import React from "react";
import { View, Dimensions } from "react-native";
import type { Node } from "react";
import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";

type Props = {
  children: Node,
  style?: StyleObj
};

const ContentPadding = ({ children, style }: Props) => {
  const paddingHorizontal = Dimensions.get("window").width >= 360 ? 16 : 8;
  return <View style={[{ paddingHorizontal }, style]}>{children}</View>;
};
ContentPadding.defaultProps = {
  style: {}
};

export default ContentPadding;
