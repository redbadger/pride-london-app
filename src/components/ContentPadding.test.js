// @flow
import React from "react";
import { Text, Dimensions } from "react-native";
import { shallow } from "enzyme";
import ContentPadding from "./ContentPadding";

describe("ContentPadding", () => {
  it("renders correctly", () => {
    expect(
      shallow(
        <ContentPadding>
          <Text>Something</Text>
        </ContentPadding>
      )
    ).toMatchSnapshot();
  });

  it("adds small padding for small devices", () => {
    const dimensionsMock = jest
      .spyOn(Dimensions, "get")
      .mockImplementation(() => ({ width: 320 }));

    const output = shallow(
      <ContentPadding>
        <Text>Something</Text>
      </ContentPadding>
    );

    expect(output.props().style[0].paddingHorizontal).toBe(8);

    dimensionsMock.mockRestore();
  });

  it("adds large padding for large devices", () => {
    const dimensionsMock = jest
      .spyOn(Dimensions, "get")
      .mockImplementation(() => ({ width: 360 }));

    const output = shallow(
      <ContentPadding>
        <Text>Something</Text>
      </ContentPadding>
    );

    expect(output.props().style[0].paddingHorizontal).toBe(16);

    dimensionsMock.mockRestore();
  });
});
