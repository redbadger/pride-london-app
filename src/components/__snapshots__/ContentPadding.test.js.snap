// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPadding renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 16,
      },
      Object {},
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Something
  </Text>
</View>
`;
